### PR TITLE
feat: re-worked content flow

### PR DESCRIPTION
**Pull request by marbemac**
Closes https://github.com/stoplightio/platform-internal/issues/9232. Result of collab between Plaid & Nauman.

Also yalc'd and manually verified inside of platform.

## Integrate JSV v4.4

Full release notes over here: https://github.com/stoplightio/json-schema-viewer/releases/tag/v4.4.0

Summary:

<details>
  <summary>Path crumbs bar that makes it easier to understand where a given property "lives" within the context of it's schema.</summary>
  
</details>

<details>
  <summary>Schema tree UI guidelines: These guidelines help visualize how deep one is in large schemas, and connect child properties at the same level together visually.</summary>
  
</details>

<details>
  <summary>Show more for long property descriptions.</summary>
  
</details>

## Highlight HTTP method & URL on operation pages

Make http operation method + url front and center at the top of operation pages.

<details>
  <summary>Example:</summary>
  Screen Shot 2021-12-02 at 12 19 09 AM
</details>

## Improve readability of HTTP operation pages

- Reduce the number of indented levels of UI across the board.
- Improve the consistency of spacing between elements.

An example:

<details>
  <summary>Before:</summary>
      <a href="https://elements-demo.stoplight.io/?spec=https://raw.githubusercontent.com/stoplightio/Public-APIs/master/reference/zoom/openapi.yaml#/operations/accounts">https://elements-demo.stoplight.io/?spec=https://raw.githubusercontent.com/stoplightio/Public-APIs/master/reference/zoom/openapi.yaml#/operations/accounts</a>

Screen Shot 2021-12-02 at 12 20 24 AM
</details>

<details>
  <summary>After:</summary>
      <a href="https://deploy-preview-1897--stoplight-elements-demo.netlify.app/?spec=https://raw.githubusercontent.com/stoplightio/Public-APIs/master/reference/zoom/openapi.yaml#/operations/accounts">https://deploy-preview-1897--stoplight-elements-demo.netlify.app/?spec=https://raw.githubusercontent.com/stoplightio/Public-APIs/master/reference/zoom/openapi.yaml#/operations/accounts</a>

Screen Shot 2021-12-02 at 12 20 53 AM
</details>

## Improved operation page markdown description <-> reference docs consistency

Adjustments such that an operation's reference docs (request, responses, etc) feel more consistent with longer markdown descriptions. 

An example:

<details>
  <summary>Before:</summary>
    <a href="https://elements-demo.stoplight.io/?spec=https://raw.githubusercontent.com/digitalocean/openapi/main/specification/DigitalOcean-public.v2.yaml#/operations/list_all_volumes">https://elements-demo.stoplight.io/?spec=https://raw.githubusercontent.com/digitalocean/openapi/main/specification/DigitalOcean-public.v2.yaml#/operations/list_all_volumes</a>

  Screen Shot 2021-12-02 at 12 03 13 AM
</details>

<details>
  <summary>After:</summary>
  <a href="https://deploy-preview-1897--stoplight-elements-demo.netlify.app/?spec=https://raw.githubusercontent.com/digitalocean/openapi/main/specification/DigitalOcean-public.v2.yaml#/operations/list_all_volumes">https://deploy-preview-1897--stoplight-elements-demo.netlify.app/?spec=https://raw.githubusercontent.com/digitalocean/openapi/main/specification/DigitalOcean-public.v2.yaml#/operations/list_all_volumes</a>

  Screen Shot 2021-12-02 at 12 07 32 AM
</details>

## Improvements to response tabs UI

- Leverage "pill" appearance tabs to more clearly indicate that they are tabs.
- Color code the tabs according to their response code for easier at a glance scanning

<details>
  <summary>Example:</summary>

  
</details>

## Other

- Refactor to leverage JSV for params rather than the previous custom parameter component. Less code to maintain, and easier to keep UX/UI consistent across params and full blown schemas.
- Adjust font size of right panel code renderers consistent with UI base size (12px), rather than the large font size that is meant for when code editor is rendered in context of markdown/prose (16px)
- Bump `json-schema-tree` to [2.1.3](https://github.com/stoplightio/json-schema-tree/releases/tag/v2.1.3) to take advantages of improvements to `allOf` merging on certain complex schema cases.

### CODE REVIEW

**Comment by mnaumanali94** [comment]
These changes are based on feedback from users who either refused to move to V7 or reluctantly did. This would make it easier for us to get everybody on v7 without pushback and remove v6 hopefully 🤞🏼 

I'd suggest the following for this: 
- Lets review the code and get it into elements
- Make a PR in platform internal, where I and somebody from Pierogi can thoroughly test it before getting it into Prod

**Comment by marbemac** [comment]
👍 and just note that I have already tested it pretty thoroughly in platform (although not "prod" platform of course).

File: packages/elements-core/src/components/Docs/HttpOperation/Body.tsx [hunk 1]
```diff
@@ -1,20 +1,28 @@
...
+import { SectionSubtitle } from '../Sections';
 
 export interface BodyProps {
   body: IHttpOperationRequestBody;
   onChange: (requestBodyIndex: number) => void;
 }
 
-export const Body = ({ body: { contents = [], description }, onChange }: BodyProps) => {
+export const isBodyEmpty = (body?: BodyProps['body']) => {
+  if (!body) return false;
```
Comment (mpodlasin) [inline packages/elements-core/src/components/Docs/HttpOperation/Body.tsx:18]: So not having body means that body is *not* empty? 🤔 Kind of counterintuitive 😅

**Comment by mpodlasin** [comment]
I mean, what can I say. I believe this looks stunning and much, much better than the previous version. <3

File: packages/elements-core/src/components/Docs/HttpOperation/Body.tsx (hunk 1 above)
Comment (marbemac) [reply packages/elements-core/src/components/Docs/HttpOperation/Body.tsx:18]: lol, I'm even confused myself just reading this again - will look at it after my meetings 😅

**Comment by wmhilton** [comment]
Loving 99.9% of this!

I have one complaint / regression - I can’t tell which `required` on the right goes with the property on the left because the line separators are gone:

image

**Comment by mnaumanali94** [comment]
> Loving 99.9% of this!
> 
> I have one complaint / regression - I can’t tell which `required` on the right goes with the property on the left because the line separators are gone:

:100: I think required should be beside the type and not all the way to the right

**Review by mpodlasin — APPROVED** [review]
My comment was addressed, so I am approving.

File: packages/elements-core/src/utils/string.ts [hunk 2]
```diff
@@ -1,3 +1,11 @@
...
 
 export const caseInsensitivelyEquals = curry((a: string, b: string) => a.toUpperCase() === b.toUpperCase());
+
+export function slugify(name: string) {
+  return name
+    .replace(/\/|{|}|\s/g, '-')
+    .replace(/-{2,}/, '-')
+    .replace(/^-/, '')
+    .replace(/-$/, '');
+}
```
Comment (P0lip) [inline packages/elements-core/src/utils/string.ts:11]:
```suggestion
export function slugify(name: string) {
  return name
    .replace(/[\/{}\s-]+/g, '-')
    .replace(/^-|-$/g, '')
}
```
wouldn't this work?

Comment (marbemac) [reply packages/elements-core/src/utils/string.ts:11]: Not sure, I just moved this function - didn't write it originally. I'll check though.

Comment (P0lip) [reply packages/elements-core/src/utils/string.ts:11]: ah, I guess we have it in Studio, fair enough then.

**Review by P0lip — APPROVED** [review]
I like it

File: packages/elements-core/src/components/TryIt/TryIt.tsx [hunk 3]
```diff
@@ -173,7 +176,7 @@ export const TryIt: React.FC<TryItProps> = ({
       let response: Response | undefined;
       try {
         response = await fetch(...request);
-      } catch (e) {
+      } catch (e: any) {
```
[the remark this answers is not in the document]
Comment (marbemac) [reply packages/elements-core/src/components/TryIt/TryIt.tsx:179]: Implicitly unknown - this is actually a broader issue in many of our codebases now due to a typescript change (forget which version did it). Can leave as is though, was just annoying me that the file was red in my IDE.

Screen Shot 2021-12-03 at 1 24 23 PM

**Pull request merged**